src/main/java/frc/robot/config/DynamicRobotConfig.java

```diff
@@ -0,0 +1,100 @@
+package frc.robot.config;
+
+import edu.wpi.first.wpilibj.Preferences;
+import java.util.logging.Logger;
+
+/**
+ * The DynamicRobotConfig class keeps track of periodically changing constants on the robot
+ *
+ * <p>This should only have offsets that would change with a mechanical change. There should be no
+ * IDs, Conversions, or setpoint information in this config.
+ */
+public class DynamicRobotConfig {
+  private static DynamicRobotConfig dynRobotConfig;
+
+  // Keys to load values from, shall not be modified under any circumstance
+  private static final String frontLeftOffset_key = "front Left Offset";
+  private static final String frontRightOffset_key = "front Right Offset";
+  private static final String backRightOffset_key = "back Right Offset";
+  private static final String backLeftOffset_key = "back Left Offset";
+
+  // Contains every key that is used
+  // if one is missing it won't be correctly saved, or init
+  private static final String[] allKeys = {
+    frontLeftOffset_key, frontRightOffset_key, backRightOffset_key, backLeftOffset_key
+  };
+
+  public final double frontLeftOffset;
+  public final double frontRightOffset;
+  public final double backRightOffset;
+  public final double backLeftOffset;
+
+  public TunerConstants getTunerConstants() {
+    return new TunerConstants(frontLeftOffset, frontRightOffset, backLeftOffset, backRightOffset);
```

Comment:
Better to create a TunerConstants class member to return here. Otherwise we get a new instance each time this method is called.